File: primeDivider.java
public class primeDivider {
    public static void main(String[] args) {
        int n = 1136;
        System.out.print(n + " = ");
        factorize(n);
    }
    public static void factorize(int n) {
        if (n <= 1) {
            System.out.println(n);
            return;
        }

        boolean first = true;
        int count = 0;
        while (n % 2 == 0) {
            n /= 2;
            count++;
        }
        if (count > 0) {
            printFactor(2, count, first);
            first = false;
        }
        for (int i = 3; i * i <= n; i += 2) {
            count = 0;
            while (n % i == 0) {
                n /= i;
                count++;
            }
            if (count > 0) {
                printFactor(i, count, first);
                first = false;
            }
        }
        if (n > 1) {
            printFactor(n, 1, first);
        }
    }
    private static void printFactor(int base, int exp, boolean first) {
        if (!first) {
            System.out.print(" * ");
        }
        System.out.print(base);
        if (exp > 1) {
            System.out.print("^" + exp);
        }
    }
}

File: prime_divider.py
# prime_divider.py

def factorize(n):
    if n <= 0:
        raise ValueError("n must be positive")
    if n == 1:
        print(n)
        return

    first = True

    count = 0
    while n % 2 == 0:
        n //= 2
        count += 1
    if count:
        print_factor(2, count, first)
        first = False

    i = 3
    while i * i <= n:
        count = 0
        while n % i == 0:
            n //= i
            count += 1
        if count:
            print_factor(i, count, first)
            first = False
        i += 2

    if n > 1:
        print_factor(n, 1, first)

def print_factor(base, exp, first):
    if not first:
        print(" * ", end="")
    print(base, end="")
    if exp > 1:
        print(f"^{exp}", end="")

if __name__ == '__main__':
    n = int(input("Give me a number: "))
    print(f"{n} = ", end="")
    factorize(n)
    print()